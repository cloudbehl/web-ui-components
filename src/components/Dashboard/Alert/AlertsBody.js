import React from 'react';
import PropTypes from 'prop-types';

export const AlertsBody = ({ children }) => <div className="kubevirt-alert__alerts-body">{children}</div>;

AlertsBody.propTypes = {
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]).isRequired,
};
