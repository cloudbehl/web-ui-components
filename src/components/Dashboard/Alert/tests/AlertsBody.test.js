import React from 'react';
import { shallow } from 'enzyme';

import { AlertsBody } from '../AlertsBody';
import { default as AlertsBodyFixtures } from '../fixtures/AlertsBody.fixture';

const testAlertsBody = () => <AlertsBody {...AlertsBodyFixtures.props} />;

describe('<AlertsBody />', () => {
  it('renders correctly', () => {
    const component = shallow(testAlertsBody());
    expect(component).toMatchSnapshot();
  });
});
