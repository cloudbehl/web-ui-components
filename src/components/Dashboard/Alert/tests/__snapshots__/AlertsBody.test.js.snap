// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlertsBody /> renders correctly 1`] = `
<div
  className="kubevirt-alert__alerts-body"
>
  <AlertItem
    alert={
      Object {
        "annotations": Object {
          "message": "Critical alert",
        },
        "labels": Object {
          "alertname": "fooName",
          "severity": "critical",
        },
      }
    }
    key="1"
  />
  <AlertItem
    alert={
      Object {
        "annotations": Object {
          "message": "unknown type alert",
        },
        "labels": Object {
          "alertname": "fooName",
          "severity": "unknown",
        },
      }
    }
    key="2"
  />
  <AlertItem
    alert={
      Object {
        "annotations": Object {
          "message": "20% of the metrics targets are down.",
        },
        "labels": Object {
          "alertname": "TargetDown",
          "severity": "warning",
        },
      }
    }
    key="3"
  />
</div>
`;
